File: 205-IsomorphicStrings/205-IsomorphicStrings.py
# Last updated: 07/04/2026, 11:40:25
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        ds , dt = {}, {}

        for i ,j in zip(s,t) : 
            c1 = i in ds and ds[i] != j 
            c2 = j in dt and dt[j] != i 

            if c1 or c2 :
                return False
            ds[i] = j
            dt[j]  = i 

        return True 


            

File: 205-IsomorphicStrings/205-IsomorphicStrings.cpp
// Last updated: 07/04/2026, 11:40:23
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        
        map<char,char> st;
        map<char,char> ts;
        if ( s.size() != t.size()) return false;
        int n = s.size();

        
        for (int i = 0 ;i<n;i++){

            if ( st.count(s[i])){
                if (st[s[i]] != t[i]) return false;
            }else{
                st[s[i]] = t[i];
            }

            if ( ts.count(t[i])){
                if ( ts[t[i]] != s[i]){
                    return false;
                }
            }
            else{
                ts[t[i]] = s[i];
            }




        }
        return true;

    }
};